lib/elastic_apm/opentracing.rb

```diff
@@ -0,0 +1,340 @@
+        @span = span
+        @scope_stack = scope_stack
+        @finish_on_close = finish_on_close
+      end
+
+      attr_reader :span
+
+      def elastic_span
+        span.elastic_span
+      end
+
+      def close
+        @span.finish if @finish_on_close
+        @scope_stack.pop
+      end
+    end
+
+    # @api private
+    class ScopeStack
+      KEY = :__elastic_apm_ot_scope_stack
+
+      def push(scope)
+        scopes << scope
+      end
+
+      def pop
+        scopes.pop
+      end
+
+      def last
+        scopes.last
+      end
+
+      private
+
+      def scopes
+        Thread.current[KEY] ||= []
+      end
+    end
+
+    # @api private
+    class ScopeManager
+      def initialize
+        @scope_stack = ScopeStack.new
+      end
+
+      def activate(span, finish_on_close: true)
+        return active if active && active.span == span
+        scope = Scope.new(span, @scope_stack, finish_on_close: finish_on_close)
+        @scope_stack.push scope
+        scope
+      end
+
+      def active
+        @scope_stack.last
+      end
+    end
+
+    # rubocop:disable Metrics/ClassLength
+    # A custom tracer to use the OpenTracing API with ElasticAPM
+    class Tracer
+      def initialize
+        @scope_manager = ScopeManager.new
+      end
+
+      attr_reader :scope_manager
+
+      def active_span
+        scope_manager.active&.span
+      end
+
+      # rubocop:disable Metrics/MethodLength, Metrics/ParameterLists
+      def start_active_span(
+        operation_name,
+        child_of: nil,
+        references: nil,
+        start_time: Time.now,
+        tags: {},
+        ignore_active_scope: false,
+        finish_on_close: true,
+        **
+      )
+        span = start_span(
+          operation_name,
+          child_of: child_of,
+          references: references,
+          start_time: start_time,
+          tags: tags,
+          ignore_active_scope: ignore_active_scope
+        )
+        scope = scope_manager.activate(span, finish_on_close: finish_on_close)
+
+        if block_given?
+          begin
+            yield scope
+          ensure
+            scope.close
+          end
+        end
+
+        scope
+      end
+      # rubocop:enable Metrics/MethodLength, Metrics/ParameterLists
+
+      # rubocop:disable Metrics/MethodLength, Metrics/ParameterLists
+      # rubocop:disable Metrics/AbcSize
+      def start_span(
+        operation_name,
+        child_of: nil,
+        references: nil,
+        start_time: Time.now,
+        tags: {},
+        ignore_active_scope: false,
+        **
+      )
+        span_context = prepare_span_context(
+          child_of: child_of,
+          references: references,
+          ignore_active_scope: ignore_active_scope
+        )
+
+        if span_context
+          trace_context =
+            span_context &&
+            span_context.respond_to?(:trace_context) &&
+            span_context.trace_context
+        end
+
+        elastic_span =
```

Comment:
At this point you would need to consider `ignore_active_span`.

From [opentracing-ruby](https://github.com/opentracing/opentracing-ruby):
>  If there is an active Scope, it will act as the parent to any newly started
>  Span unless ignore_active_scope: true is passed to start_span or
>  start_active_span.

From what I understand this means you need to start a new root span if `ignore_active_span` is set, which would either be a new transaction, or a new span that has the `current_transaction` as `parent`. Did you check with the other agent devs how this is interpreted (as `transactions` are an elastic specific concept).

Reply:
According to https://github.com/elastic/apm-agent-ruby/pull/273/files#r243277268 you were thinking of not supporting `ignore_active_scope`, but I haven't seen any documentation or log entries around this. What is your decision on this?